MobileWallet/Screens/Home/TransactionHistory/TxTableViewModel.swift

```diff
@@ -207,29 +207,86 @@ class TxTableViewModel: NSObject {
         }
     }
 
-    private func downloadGif(gifID: String) {
+    func tryDownloadGif() {
+        if gifID == nil || gif != nil { return }
         gifDownloadFailed = false
-        let downloadOperation = GiphyCore.shared.gifByID(gifID) { (response, error) in
-            guard error == nil else {
-                return TariLogger.error("Failed to load gif", error: error)
+        TxGifManager.shared.downloadGif(gifID: gifID!) { [weak self] (result) in
+            switch result {
+            case .success(let media): self?.gif = media
+            case .failure: self?.gifDownloadFailed = true
             }
-            let media = response?.data
-            TxTableViewModel.cachedMedia[gifID] = media
-            self.shouldUpdateCellSize = true
-            self.gif = media
         }
+    }
 
-        Timer.scheduledTimer(withTimeInterval: 10.0, repeats: false) { (timer) in
-            timer.invalidate()
-            if self.gif != nil { return }
-            downloadOperation.cancel()
-            self.gifDownloadFailed = true
+    func cancelDowloadGif() {
+        if gifID == nil || gif != nil { return }
+        TxGifManager.shared.cancelDownloadGif(gifID: gifID!)
+    }
+
+    override func isEqual(_ object: Any?) -> Bool {
+        guard let object = object as? TxTableViewModel else { return false }
+        return object.id == id
+    }
+}
+
+class TxGifManager {
```

Comment:
A model class is just to hold data. This looks like a UI operation so you can think about refactoring it to somewhere else, preferably a class that's more related to the UI process.